apps/03TicTacToe.js

```diff
@@ -21,32 +32,94 @@ function printBoard() {
     console.log('2 ' + board[2].join(' | '));
 }
 
-function horizontalWin() {
+function horizontalWin() { //checks for three Xs or Os in horiz. row
     // Your code here
+    if(
+    (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn)
+    ||
+    (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn)
+    ||
+    (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)
+  ) {
+    return true; //returns that horizontal win is true
+   }
+  else
+  {
+    return false; //returns that horizontal win is false
+  }
 }
 
-function verticalWin() {
-    // Your code here
+function verticalWin() { //checks for Xs and Os in vert. row
+    if(
+    (board [0][0] === playerTurn && board [1][0] === playerTurn && board[2][0] === playerTurn)
+    ||
+    (board [0][1] === playerTurn && board [1][1] === playerTurn && board[2][1] === playerTurn)
+    ||
+    (board [0][2] === playerTurn && board [1][2] === playerTurn && board[2][2] === playerTurn)
+) {
+  return true; //returns that verticalWin is true
+}
+  else
+{
+  return false; //returns that verticalWin is false
+  }
 }
 
-function diagonalWin() {
-    // Your code here
+function diagonalWin() { //checks for three Xs and Os in diag
+    if(
+    (board [0][0] === playerTurn && board [1][1] === playerTurn && board[2][2] === playerTurn)
+    ||
+    (board [0][2] === playerTurn && board [1][1] === playerTurn && board[2][0] === playerTurn)
+  ) {
+    return true; //returns that diagonalWin is true
+  }
+  else
+  {
+    return false; //returns that diagonalWin is false
+  }
 }
 
-function checkForWin() {
-    // Your code here
+function checkForWin() { //checks if horizontalWin et al are true.
+    if(horizontalWin() || verticalWin() || diagonalWin()) {
+      return true;     //if one is true, it returns true.
+    }
+    else {
+      {
+        return false; //otherwise, it returns false.
+      }
+    }
 }
 
 function ticTacToe(row, column) {
-    // Your code here
+
+    // if(wrongEntry(row,column)) {
+    //   getPrompt();
+    // }
+    // else {
+    if (board[row][column] === "X" || board[row][column] === "O" ) {
+      console.log("That spot is taken! Try again")
+    }
+     else {
+      board[row][column] = playerTurn; //takes the row/column input and makes it equal player turn
+      if(checkForWin()) { //if checkForWin is true...
+        printBoard(); //it prints the board...
+        console.log('Player ' + playerTurn + ' Won!');//and says the person won
+      }
+      else { //otherwise it checks for an X and makes it an O, and if it's not an O, makes it an X (below)
+      playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+      //acts person for the next entry, since no one won
+    }
+  }
```

Comment:
Work on the indentations here since you have a if/else statement _nested inside_ another if/else statement. Sorry for creating more work for you here lol